### ISSUE TITLE
Support for 1D datasets with N signal value and N+1 X coords in NXdata

### ISSUE THREAD

**Issue by t20100** [silx-kit/silx#3275]
Acquisition can produce N Y for N+1 X values, the idea would be to support that.
It can be either displayed as a curve with points at `0.5 (X[i] + x[i+1])`... or with a histogram?

**Comment by vallsv** [comment]
It would be good to identify properly what sub module you talk about.

I guess you talk about `silx view`? maybe NXdata?

You talk about acquisition. In this case i think the data have to be properly flagged to know how to interpret the data. I don't think it's a good idea to infer this kind of things. Or maybe you talk about a kind of fallback? (in this case it's better to cut the data i think)

If you talk about bliss acquisition. That's something which is not really recommanded (not the same amount of points). I would say, it means, the scan was not properly designed. If it's mandatory for the scientists it would be happy to have a little more feedback here, cause i will have to handle the same problems on my side.

**Comment by vasole** [comment]
@vallsv 

The facts are:

- Nexus considers acceptable to save intervals (start-stop) and a signal associated to that interval.
- Bliss continuous scans do save N+1 positions for acquisitions of N points.

**Comment by vallsv** [comment]
> Bliss continuous scans do save N+1 positions for acquisitions of N points.

In fact there is no "bliss continuous scans" in sense each beamline have to design it's own. That is why i think in your case there is a problem.

We have talked about this N+1 points for the axis with few BCU people. And no one wanted to go into this direction.

**Comment by vallsv** [comment]
Oh,

Which make me remember that sometimes you could have 2 channels for the same axis result. The original one (with N+1) and the filtered one (N values). The N+1 is there for archiving, in case you want to play with the raw data.

Anyway, that's really case by case.

**Comment by vasole** [comment]
> Which make me remember that sometimes you could have 2 channels for the same axis result. The original one (with N+1)
> and the filtered one (N values). The N+1 is there for archiving, in case you want to play with the raw data.

Exactly. That is the case I was referring to.

**Comment by vallsv** [comment]
I see, but in this case,  the NXdata should not contain the channel with N+1. Except a problem on the scan.

**Comment by vasole** [comment]
No, at our side we can deal with it if we generate things "properly".

The question becomes if we want to support the apparently valid NeXus case of N points against N+1 points. 

I see an easy solution with the 0.5 * (X[i] + X[i+1]) approach.

**Comment by t20100** [comment]
Is there a public available Nexus discussion/doc on this?
If it's Nexus compliant there is no need to discuss it.

Anyway, I would see it as something generic since it is the only way to describe an histogram.
